python/pyarrow/types.py

```diff
@@ -44,209 +44,359 @@
 def is_null(t):
     """
     Return True if value is an instance of a null type.
+
+    Parameters
+    ----------
+    t : DataType
+        type to check
```

Comment:
To be honest, this is the kind of example of a rather pointless description ("type to check" does not say anything very useful). Can we just omit it?

Reply:
I think that parameter descriptions are optional, I'll remove the descriptions and leave only the type.

Reply:
verified with `$ archery numpydoc --allow-rule PR01` didn't complain